pkg/client/cache/reflector_test.go

```diff
@@ -375,7 +376,24 @@ func TestReflectorResync(t *testing.T) {
 	}
 	resyncPeriod := 1 * time.Millisecond
 	r := NewReflector(lw, &api.Pod{}, s, resyncPeriod)
-	r.ListAndWatch(stopCh)
+	clock := util.NewFakeClock(time.Now())
+	r.clock = clock
+
+	go r.ListAndWatch(stopCh)
+
+	// advance the clock until the reflector stops
+Loop:
```

Comment:
@ncdc https://github.com/kubernetes/kubernetes/pull/26013 this simple fix might work.